### PR TITLE
Test lustre setup

### PR DESCRIPTION
**Pull request by awslitvit**
### Description of changes
* Added ChefSpec tests for lustre:setup
* Kitchen tests moved from the root cookbook to `aws-parallelcluster-common`

### Tests
* ChefSpec tests 
* Kitchen test `lustre_installation` manually run on EC2 (all the supported OSes)

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-common/kitchen.resources.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Allows to test single recipes
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-common/kitchen.resources.yml:1]: question, non blocking: probably not in the scope of this change, but we should move also other tests here, like the efa/efs ones right?

File: cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        .with(retry_delay: 5)
+
+      is_expected.to run_execute('yum-config-manager_skip_if_unavail')
+        .with(command: "yum-config-manager --setopt=\*.skip_if_unavailable=1 --save")
+
+      is_expected.to install_package(%w(kmod-lustre-client lustre-client dracut))
+        .with(retries: 3)
+        .with(retry_delay: 5)
+
+      is_expected.to install_kernel_module("lnet")
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb:152]: issue, blocking, this should be also on 7.5 and 7.6

File: cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    end
+  end
+
+  context "on centos 7.7 or higher" do
+    let(:chef_run) do
+      runner = ChefSpec::Runner.new(
+        platform: 'centos', version: '7',
+        step_into: ['lustre']
+      ) do |node|
+        node.override['cluster']['platform_version'] = "7.7"
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb:129]: nitpick, probably overkill, but we could test that there are some unsupported minor versions

```
  case node['platform_version'].to_i
  when 7
    os_minor_version = '7' if kernel_patch_version[1] >= '1062'
    os_minor_version = '8' if kernel_patch_version[1] >= '1127'
    os_minor_version = '9' if kernel_patch_version[1] >= '1160'
  else
    raise "CentOS version #{node['platform_version']} not supported."
  end

```

File: cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+      it 'installs lustre packages from repository and installs kernel module lnet' do
+        is_expected.to add_apt_repository('fsxlustreclientrepo')
+          .with(uri: 'https://fsx-lustre-client-repo.s3.amazonaws.com/ubuntu')
+          .with(components: %w(main))
+          .with(key: ["https://fsx-lustre-client-repo-public-keys.s3.amazonaws.com/fsx-ubuntu-public-key.asc"])
+          .with(retries: 3)
+          .with(retry_delay: 5)
+
+        is_expected.to periodic_apt_update('')
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb:244]: praise: cool! I didn't knew `periodic` was the default, wondering if we should explicit it somewhere

File: cookbooks/aws-parallelcluster-common/test/common/libraries/os_properties.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
+#
+# Check if we are running in a virtualized environment
```
Comment (eantonin) [inline cookbooks/aws-parallelcluster-common/test/common/libraries/os_properties.rb:2]: question, non blocking: is this file repeated for all the cookbooks?

**Review by eantonin — CHANGES_REQUESTED** [review]

File: cookbooks/aws-parallelcluster-common/kitchen.resources.yml (hunk 1 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/kitchen.resources.yml:1]: Yes, but I considered it out of scope. Will do in the following PRs.

File: cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb (hunk 2 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb:152]: Correct, thanks!

File: cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb (hunk 3 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb:129]: Thanks, will do in future PRs.

File: cookbooks/aws-parallelcluster-common/test/common/libraries/os_properties.rb (hunk 5 above)
Comment (awslitvit) [reply cookbooks/aws-parallelcluster-common/test/common/libraries/os_properties.rb:2]: Yes, for now. Going to remove it in a future PR.

**Review by eantonin — APPROVED** [review]
LGTM

**Pull request merged**